pkg/machinery/resources/hardware/processor.go

```diff
@@ -0,0 +1,78 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package hardware
+
+import (
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+)
+
+//go:generate deep-copy -type ProcessorSpec -type MemorySpec -header-file ../../../../hack/boilerplate.txt -o deep_copy.generated.go .
+
+// ProcessorType is type of Processor resource.
+const ProcessorType = resource.Type("Processors.hardware.talos.dev")
+
+// ProcessorInfo resource holds node ProcessorInfo information.
+type ProcessorInfo = typed.Resource[ProcessorSpec, ProcessorRD]
```

Comment:
nit: 
```suggestion
type Processor = typed.Resource[ProcessorSpec, ProcessorRD]
```

we can call it like the resource type?